components/SampleAddressInput/SampleAddressInput.tsx

```diff
@@ -0,0 +1,68 @@
+/**
+ * @author Felix Hildebrandt <fhildeb>
+ */
+
+import { useContext } from 'react';
+import { NetworkContext } from '../../contexts/NetworksContext';
+
+enum AddressType {
+  UP = 'UP',
+  Asset = 'Asset',
+}
+
+interface Props {
+  inputId: string;
+  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
+}
+
+const SampleAddressInput: React.FC<Props> = ({ inputId, onChange }) => {
+  const { network } = useContext(NetworkContext);
+
+  const changeInputAddress = (type: AddressType) => {
+    let address;
+    if (type === AddressType.UP) {
+      if (network.name === 'MAINNET') {
+        // TODO: Update to Mainnet Profile
+        address = '0x9139def55c73c12bcda9c44f12326686e3948634';
+      } else if (network.name === 'TESTNET') {
+        address = '0x9139def55c73c12bcda9c44f12326686e3948634';
+      }
+    } else if (type === AddressType.Asset) {
+      if (network.name === 'MAINNET') {
+        // TODO: Update to Mainnet Asset
+        address = '0x6395b330F063F96579aA8F7b59f2584fb9b6c3a5';
+      } else if (network.name === 'TESTNET') {
+        address = '0x6395b330F063F96579aA8F7b59f2584fb9b6c3a5';
```

Comment:
Could you define these addresses in a const or global files - or at the very list at the top of this file.

It is important to give names to addresses as over time we don't know what is what.

```js
const SAMPLE_MAINNET_UP_ADDRESS = '';
const SAMPLE TESTNET_UP_ADDRESS = '0x9139def55c73c12bcda9c44f12326686e3948634';

// NOTE: what kind of asset btw?
const SAMPLE_MAINNET_ASSET = '';
const SAMPLE_TESTNET_ASSET = '0x6395b330F063F96579aA8F7b59f2584fb9b6c3a5';
```

Also, you are using the same addresses for testnet and mainnet - is it intended? Let's not hide little "bombs" in the code, just change the address to empty string for MAINNET. At least it is clear that it is missing.

Having a testnet address like it is right not is wrong and might lead to problems.

Reply:
Consider using a switch case